powertools-serialization/src/test/java/software/amazon/lambda/powertools/utilities/jmespath/Base64GZipFunctionTest.java

```diff
@@ -33,4 +46,24 @@ public void testPowertoolsGzip() throws IOException {
         assertThat(result.getNodeType()).isEqualTo(JsonNodeType.STRING);
         assertThat(result.asText()).isEqualTo("{  \"id\": 43242,  \"name\": \"FooBar XY\",  \"price\": 258}");
     }
+
+    @Test
+    public void testPowertoolsGdzipEmptyJsonAttribute() throws IOException {
+        JsonNode event = JsonConfig.get().getObjectMapper().readTree(this.getClass().getResourceAsStream("/custom_event_gzip.json"));
+        Expression<JsonNode> expression = JsonConfig.get().getJmesPath().compile("basket.powertools_base64_gzip('')");
+        JsonNode result = expression.search(event);
+        assertThat(result.getNodeType()).isEqualTo(JsonNodeType.STRING);
```

Comment:
Should we actually return null instead. If the element does not exist... I think you removed the null check, why?

Reply:
Yep, that was [wrong](https://github.com/aws-powertools/powertools-lambda-java/pull/1264#discussion_r1254133934) removing the null check.

In regards to what we return, I think the best option is to either throw an exception or return an optional, so that it is mandatory for the caller to do something with it. An empty string or a null can be forgotten.
Comparing between the null and the empty string, I would go for the empty string since there are less chances for an unhandled nullPointer exception.